### PR TITLE
README Improvement

### LINKED ISSUE #864: _LINTER_RULES rules doesn't exist
I'm also seeing issues when `MARKDOWN_CONFIG_FILE` is set, it is unable to find the file unless `LINTER_RULES_PATH` is also set.

Doesn't work

```yaml
      - uses: github/super-linter@v3
        env:
          VALIDATE_MARKDOWN: true
          MARKDOWN_CONFIG_FILE: docs/.markdownlint.yml
```

Works

```yaml
      - uses: github/super-linter@v3
        env:
          LINTER_RULES_PATH: '.' # From https://github.com/github/super-linter/pull/859#issuecomment-709625090
          VALIDATE_MARKDOWN: true
          MARKDOWN_CONFIG_FILE: docs/.markdownlint.yml
```

Continued from #858 & #859

> Not sure if it's #854 or #859 but, if you provide, for example, JAVASCRIPT_ES_CONFIG_FILE it still tries to look for it in the DEFAULT_RULES_LOCATION when LINTER_RULES_PATH is not set (so i guess it ignores the default?). When you set LINTER_RULES_PATH it works fine.

_Originally posted by @PhunStyle in https://github.com/github/super-linter/pull/859#issuecomment-709625090_

### LINKED ISSUE #872: Give users the ability to enable or disable stylelint's cosmiconfig functionality
Hi folks :wave:

I have a suggestion that may improve user experience for developers who are using super-linter with stylelint to lint CSS code. If implemented, the user experience when using super-linter with stylelint would be similar to what a developer experiences when they run stylelint individually without super-linter. 

Please let me know your thoughts on this. 

**Is your feature request related to a problem? Please describe.**

Yes, I noticed that super-linter is executing the ``stylelint`` binary with the ``--config`` flag by default. When executing ``stylelint`` with the ``--config`` flag, [cosmiconfig](https://stylelint.io/user-guide/configure) is disabled. Cosmiconfig is functionality that is built into stylelint that searches a repo for a stylelintrc config file, such as stylelintrc.yml, and then loads it into ``stylelint``. 

https://github.com/github/super-linter/blob/eec8ad51b52ef2622d08e6598042eb98dce3f5b2/lib/linter.sh#L1130

Since super-linter is executing ``stylelint`` with the ``--config`` flag by default, users have to tell super-linter where the stylelintrc config file lives in their repo, and what the config file's name is with the ``LINTER_RULES_PATH`` and ``CSS_FILE_NAME`` env vars. 

**Describe the solution you'd like**

1) The ability to run stylelint with cosmiconfig

    If the user sets the ``VALIDATE_CSS: true`` env var in super-linter's ``linter.yml`` without setting the ``LINTER_RULES_PATH`` and ``CSS_FILE_NAME`` env vars then execute the ``stylelint`` binary without the ``--config`` flag.

    For example, if a user wants to run ``stylelint`` with cosmiconfig enabled they would configure super-linter's ``linter.yml`` with only the following env var:
    ```yaml
    VALIDATE_CSS: true
    ```

2) The ability to run stylelint without cosmiconfig

    If the ``VALIDATE_CSS: true``, ``LINTER_RULES_PATH``, and ``CSS_FILE_NAME`` env vars are set by the user in ``linter.yml`` then have super-linter execute ``stylelint`` as it does now, which is with the ``--config`` flag. 

    For example, if a user wants to run ``stylelint`` without cosmiconfig enabled they would configure ``linter.yml`` with the following env vars:
    ```yaml
    LINTER_RULES_PATH: /
    VALIDATE_CSS: true
    CSS_FILE_NAME: .stylelintrc.yml
    ```

**Describe alternatives you've considered**
If this functionality is not implemented, it may make sense to add a note to the README file letting users know that (a) stylelint's cosmiconfig is disabled by default in super-linter, and (b) the ``LINTER_RULES_PATH`` and ``CSS_FILE_NAME`` env vars should be configured if a user plans on using a stylelintrc config file with super-linter.

**Additional context**
If implemented, this may help folks who are configuring super-linter for the first time avoid #685 when configuring super-linter to lint CSS code.  

**Note**
None

### PR DESCRIPTION
**Pull request by josephmarino**
Hi folks,

I added some clarity to the README that may help developers new to super-linter avoid running into issues #864 and #685. I added a "Using your own rules" section under the "Template rules" section in the README. This new section talks about how users can configure super-linter to use their own rules files if their rules files live outside of a .github/linters directory in their repo. 

I hope that this helps, or is at least a starting point that turns into documentation that helps with this issue. If anyone has any suggestions feel free to add them 😃 

cc: @pdelre, @PhunStyle, @ferrarimarco

#Fixes #864 
#Fixes #872 

### Author/Contributor
- [X] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by PhunStyle** [comment]
Thank you! Not sure if this issue is still present but when I commented on https://github.com/github/super-linter/issues/864#issuecomment-711173905 this, I had a problem when where I didn't set LINTER_RULES_PATH it didn't work, so I had to still set it to the default you specify in the documentation (.github/linters/) and only after I did that, it picked up my custom config.

**Comment by pdelre** [comment]
Looks good. I believe as a new user, I would have found this information.

File: README.md [hunk 1]
```diff
@@ -296,7 +297,11 @@ You can use the **GitHub** **Super-Linter** _with_ or _without_ your own persona
 - Copy **any** or **all** template rules files from `TEMPLATES/` into your repository in the location: `.github/linters/` of your repository
   - If your repository does not have rules files, they will fall back to defaults in [this repository's `TEMPLATE` folder](https://github.com/github/super-linter/tree/master/TEMPLATES)
 
-## Disabling rules
+### Using your own rules files
```
Comment (GaboFDC) [inline README.md:300]:
```suggestion
## Using your own rules files
```

File: README.md [hunk 2]
```diff
@@ -296,7 +297,11 @@ You can use the **GitHub** **Super-Linter** _with_ or _without_ your own persona
...
+
+If your repository contains your own rules files that live outside of a ``.github/linters/`` directory, you will have to tell Super-Linter where your rules files are located in your repository, and what their file names are. To learn more, see [Using your own rules files](docs/using-rules-files.md).
+
+### Disabling rules
```
Comment (GaboFDC) [inline README.md:304]:
```suggestion
## Disabling rules
```

File: docs/using-rules-files.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Using your own rules files
+
+If your repository contains your own rules files that live outside of a ``.github/linters/`` directory, you will have to tell Super-Linter where your rules files are located in your repository, and what their file names are.
+
+You can tell Super-Linter where your rules files are located with the ``LINTER_RULES_PATH`` ENV VAR, and you can tell Super-Linter what their file names are by using any of the ENV VARS from the table below. 
+
+| ENV VARS that can be used with LINTER_RULES_PATH |
+| --------------------------------- |
+| CSS_FILE_NAME                     |
+| DOCKERFILE_HADOLINT_FILE_NAME     |
```
Comment (GaboFDC) [inline docs/using-rules-files.md:10]: Maybe is better to refer to the table in the readme? so we don't have 2 tables to maintain, and also the list is always up to date and has some extra information

Comment (josephmarino) [reply docs/using-rules-files.md:10]: @GaboFDC Good idea; that would be a lot easier to manage. I will commit a modification shortly.

**Review by josephmarino — COMMENTED** [review]
👍

File: docs/using-rules-files.md (hunk 3 above)
Comment (josephmarino) [reply docs/using-rules-files.md]: @GaboFDC 

I just committed a revision removing the new table, please share your thoughts when you have a moment. I also added some text letting folks know that they should refer to refer to the environment variables table in the readme for a list of filename env vars that can be configured with LINTER_RULES_PATH.

**Comment by josephmarino** [comment]
Hi folks I ran into an issue on my end when committing this change, sorry about that. This was my first time committing changes to an open pull request.

**Review by admiralAwkbar — APPROVED** [review]

**Pull request merged**